### PR TITLE
[SPARK-29742][BUILD] Update checkstyle plugin's check dir scope

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Current checkstyle checking folder can't cover all folder.
Since for support multi version hive, we have some divided hive folder.
We should check it too.

### Why are the changes needed?
Fix build bug

### Does this PR introduce any user-facing change?
NO

### How was this patch tested?
NO

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
cc @srowen @wangyum @dongjoon-hyun

**Comment by wangyum** [comment]
ok to test

File: pom.xml [hunk 1]
```diff
@@ -2750,9 +2750,13 @@
           <sourceDirectories>
             <directory>${basedir}/src/main/java</directory>
             <directory>${basedir}/src/main/scala</directory>
+            <directory>v${hive.version.short}/src/main/java</directory>
```
Comment (wangyum) [inline pom.xml]: `v${hive.version.short}/src/main/java` -> `${basedir}/v${hive.version.short}/src/main/java`?

Comment (AngersZhuuuu) [reply pom.xml]: > `v${hive.version.short}/src/main/java` -> `${basedir}/v${hive.version.short}/src/main/java`?

done

**Review by srowen — APPROVED** [review]
I think it's OK, if there are few style checks the fix. The minor issue is that some of this code is copied from Hive right? so if we modified it a lot it might be harder to update. But minor changes like this are no big deal.

**Comment by AngersZhuuuu** [comment]
> I think it's OK, if there are few style checks the fix. The minor issue is that some of this code is copied from Hive right? so if we modified it a lot it might be harder to update. But minor changes like this are no big deal.

Not all copied code from hive.  sql/core module have some java code in folder v1.2.1/v2.3.5 .

**Review by wangyum — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
Yes. Right. +1, LGTM. Thank you, @AngersZhuuuu , @srowen , @wangyum .
Merged to master.

**Pull request closed without merging**